app/lib/account/backend.dart

```diff
@@ -133,16 +138,31 @@ class AccountBackend {
     return user;
   }
 
+  /// Returns the URL of the authorization endpoint used by pub site.
+  String siteAuthorizationUrl(Uri sourceUrl) {
+    return _siteAuthProvider.authorizationUrl(sourceUrl);
+  }
+
+  /// Validates the authorization [code] (specifying the same [redirectUrl]
+  /// that was used to access the code in the first place), and returns the
+  /// access token.
+  ///
+  /// Returns null on any error, or if the token is expired, or the code is not
+  /// verified.
+  Future<String> siteAuthCodeToAccessToken(String code, String redirectUrl) =>
```

Comment:
Either change `redirectUrl` to `hostname` or use a `hostname` hardcoded into the configuration...